### PR TITLE
fix: Cleanup T after early dialog dlg_end_dlg

### PR DESCRIPTION
**Pull request by vladpaiu**
**Summary**
Potential fix for https://github.com/OpenSIPS/opensips/issues/3498

**Details**
T is leaked when dialog looks up the transaction for early dialog hangup

**Solution**
Restore T after lookup ( hopefully to NULL )

**Compatibility**
Should be backwards compatible

**Closing issues**
https://github.com/OpenSIPS/opensips/issues/3498

### CODE REVIEW

**Comment by vladpaiu** [comment]
not stale

**Comment by vladpaiu** [comment]
still valid

**Comment by bogdan-iancu** [comment]
@vladpaiu , totally valid, both the report and the fix.  And I see a similar case with the same fix (already in place) in the [b2b_entities module](https://github.com/OpenSIPS/opensips/blob/master/modules/b2b_entities/dlg.c#L2460).
This `t_lookup_ident()` functions is used , via the TM API, only from dialog (your case) and from b2b_entities (case I found) modules only. It is also internally used by TM module - for some MI and inject scenarios.
My point is, in none of the case, there is no larger TM context needed, the scope of the transaction pointer is local, for couple of lines of code. Even more the, `t_lookup_ident()`  returns the found transaction as parameter .
So, bottom line - should the right fix be simply [NOT setting the T inside](https://github.com/OpenSIPS/opensips/blob/master/modules/tm/t_lookup.c#L1223) this `t_lookup_ident()` ??
By doing this, we solve in a simpler way the usage of the function via TM API (we do not need the fix here) and there is 0 impact of the TM internal usage of the function also.

**Comment by vladpaiu** [comment]
not sure if that's good enough.. as maybe in the future in some cases we need the T to be set so subsequent callbacks done after lookup_ident find the proper T set...

**Comment by bogdan-iancu** [comment]
That's the point - right now we do not have such a case; in all existing cases the transaction is actually returned and used via function param...and whatever transactional processing follows, it does not use the T...

**Comment by bogdan-iancu** [comment]
I just pushed a fix, as per [my initial comment](https://github.com/OpenSIPS/opensips/pull/3502#issuecomment-2697802294) .
